test/extended/prometheus/collection_profiles.go

```diff
@@ -97,17 +97,31 @@ var _ = g.Describe("[sig-instrumentation][OCPFeatureGate:MetricsCollectionProfil
 	})
 
 	g.AfterAll(func() {
+		shouldDeleteConfiguration := false
 		currentConfiguration, err := r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Get(tctx, operatorConfigurationName, metav1.GetOptions{})
 		o.Expect(err).To(o.BeNil())
 		if r.originalOperatorConfiguration != nil {
 			currentConfiguration.Data = r.originalOperatorConfiguration.Data
 			g.By("restoring the original configuration for the operator")
 			_, err = r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Update(tctx, currentConfiguration, metav1.UpdateOptions{})
 		} else {
+			shouldDeleteConfiguration = true
 			g.By("cleaning up the configuration for the operator as it did not exist pre-job")
 			err = r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Delete(tctx, operatorConfigurationName, metav1.DeleteOptions{})
 		}
 		o.Expect(err).To(o.BeNil())
+
+		o.Eventually(func() error {
```

Comment:
As I'm not a big user of `gomega`, I'm interested to know why this double check is needed, I'd expected the deletion +  `o.Expect(err).To(o.BeNil())` above to be sufficient, if not, it's a bug in `Delete` or `gomega`

Reply:
I wanted to ensure here that the configmap is actually deleted before the test exits.

Reply:
AFAIK, the default invocation of `Delete()` uses foreground deletion which waits for the resource to disappear before returning.
But maybe I'm missing sth.
It's just a remark though, not trying to block the merge.

Reply:
Seeing https://github.com/kubernetes/kubernetes/blob/9dd53b55f7b21deb8d7cc4f096899af8f46bd083/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L551 made me think that foreground is the default.
You're right, it defaults to background.

So maybe, instead of Getting the configmap and fail the test if it's still there (for some reason, an overwhelmed api or sth who knows), maybe we can just tell the api to use foreground via the `DeleteOptions`, in that case we give the deletion more chance/time.